src/deriver/agent/worker.py

```
@@ -0,0 +1,113 @@
+                session_name=session_name,
+                configuration=message_level_configuration,
+                observer=observer,
+                observed=observed,
+                observed_peer_card=observed_peer_card,
+            )
+
+            # Run agent loop for the message batch
+            await agent.run_loop(messages)
+
+        except Exception as e:
+            logger.error(
```

Ruff (0.14.10)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)